src/pygeosx/unitTests/dataRepositoryTests/test_pygeosxDataRepository.py

```diff
@@ -0,0 +1,181 @@
+            target_type (dtype): Expected type for the wrapper elements
+            geosx_proplem (pygeosx.problem): The problem handle
+        """
+        # yapf: enable
+        s = 0
+        val = 0
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "to_numpy"):
+            x = x.to_numpy()
+
+        if isinstance(x, np.ndarray):
+            s = np.shape(x)
+            Ia = tuple([0 for y in s])
+            val = x[Ia]
+        else:
+            s = len(x)
+            val = x
+
+        assert s == target_shape
+        assert isinstance(val, target_type)
+
+    # yapf: disable
+    @pytest.mark.parametrize('target_key',
+                             [('Events/time'),
+                              ('Events/cycle'),
+                              (node_manager + 'ReferencePosition'),
+                              (node_manager + 'ghostRank'),
+                              (cell_blocks + 'elementVolume')])
+    def test_wrapper_write(self, target_key, geosx_problem):
+        """
+        Test writing to the wrappers
+
+        Args:
+            target_key (str): Key value for the target wrapper
+            geosx_proplem (pygeosx.problem): The problem handle
+        """
+        # yapf: enable
+        # Access and record the initial value
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "set_access_level"):
+            x.set_access_level(pylvarray.MODIFIABLE, pylvarray.CPU)
+        if hasattr(x, "to_numpy"):
+            x = x.to_numpy()
+        x_init = x.copy()
+
+        # Modify the value
+        x += 1
+        del x
+
+        # Get another copy of the wrapper
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "to_numpy"):
+            x = x.to_numpy()
+
+        # Check to see if the value changed appropriately
+        dx = x_init + 1 - x
+        if isinstance(dx, np.ndarray):
+            dx = np.sum(dx)
+        dx = float(dx)
+
+        assert dx < 1e-10
+
+    # yapf: disable
+    @pytest.mark.parametrize('target_key',
+                             [(node_manager + 'ReferencePosition'),
+                              (node_manager + 'ghostRank'),
+                              (cell_blocks + 'elementVolume')])
+    def test_wrapper_read_only_error(self, target_key, geosx_problem):
+        """
+        Test writing to read-only wrappers.
+        Note: this error does not get called on certain scalar wrappers (time, cycle)
+
+        Args:
+            target_key (str): Key value for the target wrapper
+            geosx_proplem (pygeosx.problem): The problem handle
+        """
+        # yapf: enable
+        x = geosx_problem.get_wrapper(target_key).value()
+        if hasattr(x, "to_numpy"):
+            print('Test_a')
+            x = x.to_numpy()
+
+        with pytest.raises(ValueError):
+            x += 1
+
+    def test_problem_run(self, geosx_problem):
```

Comment:
This can be triggered also by the getting access to the solver and calling `solver.execute`, this other approach could be tested in a new pytest file (`test_pygeosxSolver.py`?), cf. comment in the PR.